log/zapcore_handler.go

```diff
@@ -0,0 +1,123 @@
+package log
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"istio.io/pkg/structured"
+)
+
+var (
+	toLevel = map[zapcore.Level]Level{
+		zapcore.FatalLevel: FatalLevel,
+		zapcore.ErrorLevel: ErrorLevel,
+		zapcore.WarnLevel:  WarnLevel,
+		zapcore.InfoLevel:  InfoLevel,
+		zapcore.DebugLevel: DebugLevel,
+	}
+	toZapLevel = map[Level]zapcore.Level{
+		FatalLevel: zapcore.FatalLevel,
+		ErrorLevel: zapcore.ErrorLevel,
+		WarnLevel:  zapcore.WarnLevel,
+		InfoLevel:  zapcore.InfoLevel,
+		DebugLevel: zapcore.DebugLevel,
+	}
+)
+
+func init() {
+	RegisterDefaultHandler(ZapLogHandlerCallbackFunc)
+}
+
+// ZapLogHandlerCallbackFunc is the handler function that emulates the previous Istio logging output and adds
+// support for errdict package and labels logging.
+func ZapLogHandlerCallbackFunc(
+	level Level,
+	scope *Scope,
+	ie *structured.Error,
+	msg string,
+	fields []zapcore.Field) {
+	if ie != nil {
+		fields = append(fields, zap.String("message", msg))
+		fields = append(fields, zap.String("moreInfo", ie.MoreInfo))
+		fields = append(fields, zap.String("impact", ie.Impact))
```

Comment:
should we append only if they are not empty?

Reply:
done.